lib/simplecov.rb

```diff
@@ -54,6 +54,42 @@ def start(profile = nil, &block)
       start_coverage_measurment
     end
 
+    #
+    # Collate a series of SimpleCov result files into a single SimpleCov output.
+    # You can optionally specify configuration with a block:
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"]
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"], 'rails' # using rails profile
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"] do
+    #     add_filter 'test'
+    #   end
+    #    OR
+    #   SimpleCov.collate Dir["simplecov-resultset-*/.resultset.json"], 'rails' do
+    #     add_filter 'test'
+    #   end
+    #
+    # Please check out the RDoc for SimpleCov::Configuration to find about available config options
+    #
+    def collate(result_filenames, profile = nil, &block)
+      load_profile(profile) if profile
+      configure(&block) if block_given?
+      self.running = true
+      self.pid = Process.pid
+
+      results = result_filenames.map do |filename|
+        # Re-create each included instance of SimpleCov::Result from the stored run data.
+        (JSON.parse(File.read(filename)) || {}).map do |command_name, coverage|
+          SimpleCov::Result.from_hash(command_name => coverage)
+        end
+      end.flatten
```

Comment:
Shouldn't we be able to use `flat_map` ?

Reply:
Fixed by https://github.com/colszowka/simplecov/pull/780/commits/e5600ffcb43bbd915e9839ae18b210aacd6ae209.